site/app/templates/calendar/Calendar.twig

```diff
@@ -79,11 +79,37 @@
         <div class="cal-legend">
             {% if show_all_cookie == 1 %}
                 {% for course in course_names %}
-                    <div name="{{course.course_name|replace({' ':''})}}" class="legend-color"></div>
-                    <p for="{{course.course_name|replace({' ':''})}}-color" class="legend-label">{{course.display_name}}</p>
+                <div 
+                    name="{{ course.course_name|replace({' ':''}) }}" 
+                    class="legend-color" 
+                    onclick="filter_course('{{ course.course_name }}', '{{ course.display_name }}')" 
+                    style="cursor: pointer;" 
+                    title="Click to filter course, click again to show all courses"
+                ></div>
```

Comment:
I think it would be best to change this to a checkbox, instead of trying to mimic a checkbox by styling a div.  By using a checkbox, it would be more accessible, and more consistent with the UI users are used to seeing on their respective platforms.

Reply:
I just tried using a checkbox, I think the checkbox is better if it doesnt takes a refresh and removes rest of the course from the row. I will also paste image in PR description of how it looks after clicking on a course.

Reply:
Why is the refresh necessary?

Reply:
It looks like this
https://github.com/Submitty/Submitty/assets/122962727/e66d2928-1dd5-445a-bb3c-c79ceb3ac8d9